homeassistant/components/conversation/default_agent.py

```diff
@@ -118,25 +75,119 @@ async def async_process(
         language: str | None = None,
     ) -> ConversationResult | None:
         """Process a sentence."""
-        intents = self.hass.data[DOMAIN]
-
-        for intent_type, matchers in intents.items():
-            for matcher in matchers:
-                if not (match := matcher.match(text)):
-                    continue
-
-                intent_response = await intent.async_handle(
-                    self.hass,
-                    DOMAIN,
-                    intent_type,
-                    {key: {"value": value} for key, value in match.groupdict().items()},
-                    text,
-                    context,
-                    language,
-                )
-
-                return ConversationResult(
-                    response=intent_response, conversation_id=conversation_id
-                )
+        language = language or self.hass.config.language
+        lang_intents = await self.async_get_or_load_intents(language)
+        if lang_intents is None:
+            # No intents loaded
+            return None
```

Comment:
Should we return an error when we don't support a language?

Reply:
I print a warning for now. The `_async_converse` function interprets `None` as a match being unsuccessful.

Reply:
I think that it would be good to tell the user why it didn't match. Maybe they are trying to add sentences and they are not understanding which language is being used for processing.